lib/sinatra/base.rb

```diff
@@ -1086,7 +1087,7 @@ def invoke
 
     # Dispatch a request with error handling.
     def dispatch!
-      force_encoding(@params = IndifferentHash[@request.params])
+      force_encoding(@params = @params.merge(@request.params))
```

Comment:
Is there any reason to not use the `merge!` method? It would be something like:

```ruby
force_encoding(@params.merge!(@request.params))
```

Reply:
Good point, I'll make that update 👍